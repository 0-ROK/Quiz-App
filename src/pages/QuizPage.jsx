import {
  Card,
  CardActions,
  CardContent,
  CardHeader,
  Skeleton,
} from "@mui/material";
import React, { useEffect, useState } from "react";
import { useRecoilValue } from "recoil";
import useQuiz from "../hooks/useQuiz";
import Counter from "../modules/Counter";
import QuizForm from "../modules/QuizForm";
import { resolvedQuizState } from "../states/recoilResolvedQuizState";

const MAX_COUNT = 10;

const QuizPage = () => {
  const [quiz, skipQuiz, submitQuiz] = useQuiz();

  const resolvedQuiz = useRecoilValue(resolvedQuizState);

  return (
    <div>
      <header className="App-header">
        <span>{`${resolvedQuiz.length}/${MAX_COUNT}`}</span>
        <span>{resolvedQuiz.map((quiz) => (quiz.result ? "✅" : "❌"))}</span>
        <Counter />
        <br />
        <div style={{ width: "80%", height: "40%" }}>
          <QuizForm
            quiz={quiz}
            skipQuiz={skipQuiz}
            submitQuiz={submitQuiz}
            width="100%"
            height="100%"
          />
        </div>
      </header>
    </div>
  );
};

export default QuizPage;
