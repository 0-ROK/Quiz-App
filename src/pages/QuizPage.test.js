import { render, screen } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import QuizPage from "./QuizPage";
import { BrowserRouter as Router } from "react-router-dom";

describe("<QuizPage/>", () => {
  it("is quizpage snapshot", () => {
    const utils = render(
      <RecoilRoot>
        <Router>
          <QuizPage />
        </Router>
      </RecoilRoot>
    );
    expect(utils.container).toMatchSnapshot();
  });

  it("데이터를 불러오는 동안은 문항이 skeleton으로 표시된다.", async () => {
    render(
      <RecoilRoot>
        <Router>
          <QuizPage />
        </Router>
      </RecoilRoot>
    );

    const quiz = screen.queryByText(/^\d\. .+?/);
    expect(quiz).toBe(null);

    await screen.findByText(/^\d\. .+?/);
  });

  it("데이터를 불러오는 동안은 난이도가 skeleton으로 표시된다.", async () => {
    render(
      <RecoilRoot>
        <Router>
          <QuizPage />
        </Router>
      </RecoilRoot>
    );

    const difficulty = screen.queryByDisplayValue(/\b(:?easy|medium|hard)\b/);
    expect(difficulty).toBe(null);
  });

  it("데이터를 불러오는 동안은 보기가 skeleton으로 표시된다.", async () => {
    render(
      <RecoilRoot>
        <Router>
          <QuizPage />
        </Router>
      </RecoilRoot>
    );

    const answers = screen.queryAllByDisplayValue(/^/);
    expect(answers?.length).toBe(0);

    const waitedAnswers = await screen.findAllByDisplayValue(/^/);
    expect(waitedAnswers?.length).toBe(4);
  });
});
