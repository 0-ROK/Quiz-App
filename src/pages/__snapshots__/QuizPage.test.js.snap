// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizPage/> is quizpage snapshot 1`] = `
<div>
  <div>
    <div
      class="App-header"
    >
      <span>
        ÏßÑÌñâÎèÑ : 1/10
      </span>
      <span />
      <span>
        ÏÜåÏöîÏãúÍ∞Ñ : 0Ï¥à
      </span>
      <br />
      <div
        style="width: 80%; height: 40%;"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <form>
            <div
              class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                    style="width: 80%; margin-bottom: 6px;"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                    style="width: 2%; height: 10px;"
                  />
                </span>
              </div>
            </div>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div>
                <fieldset>
                  <legend>
                    loading...
                  </legend>
                  <div
                    aria-labelledby="demo-radio-buttons-group-label"
                    class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                    role="radiogroup"
                  >
                    <span
                      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                      style="width: 20%; margin-bottom: 10px;"
                    />
                    <span
                      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                      style="width: 20%; margin-bottom: 10px;"
                    />
                    <span
                      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                      style="width: 20%; margin-bottom: 10px;"
                    />
                    <span
                      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
                      style="width: 20%; margin-bottom: 10px;"
                    />
                  </div>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
                disabled=""
                tabindex="-1"
                type="submit"
              >
                submit!
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                skipüòÖ
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
